File: Projetos/ACBrLib/Demos/C#/Shared/ACBrLib.Core/Config/PrincipalConfig.cs
namespace ACBrLib.Core.Config
{
    public sealed class PrincipalConfig<TLib> : ACBrLibConfigBase<TLib> where TLib : ACBrLibHandle
    {
        #region Constructors

        public PrincipalConfig(TLib acbrlib) : base(acbrlib, ACBrSessao.Principal)
        {
        }

        #endregion Constructors

        #region Properties

        public TipoResposta TipoResposta
        {
            get => GetProperty<TipoResposta>();
            set => SetProperty(value);
        }

        public CodResposta CodificacaoResposta
        {
            get => GetProperty<CodResposta>();
            set => SetProperty(value);
        }

        public NivelLog LogNivel
        {
            get => GetProperty<NivelLog>();
            set => SetProperty(value);
        }

        public string LogPath
        {
            get => GetProperty<string>();
            set => SetProperty(value);
        }

        #endregion Properties
    }
}

File: Projetos/ACBrLib/Demos/Java/ACBr Common/src/com/acbr/PrincipalConfig.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.acbr;

/**
 *
 * @author rften
 * @param <TLib>
 */
public final class PrincipalConfig<TLib extends ACBrLibBase> extends ACBrLibConfigBase<TLib>{
    
    public PrincipalConfig(TLib acbrlib) {
        super(acbrlib, ACBrSessao.Principal);
    }
    
    public TipoResposta getTipoResposta() throws Exception{
       return TipoResposta.valueOf(getIntProperty("TipoResposta"));
    }
    
    public void setTipoResposta(TipoResposta value) throws Exception {
        this.setIntProperty("TipoResposta", value.asInt());
    }
    
    public CodResposta getCodificacaoResposta() throws Exception{
       return CodResposta.valueOf(getIntProperty("CodificacaoResposta"));
    }
    
    public void setCodificacaoResposta(CodResposta value) throws Exception {
        this.setIntProperty("CodificacaoResposta", value.asInt());
    }
    
    public NivelLog getLogNivel() throws Exception{
       return NivelLog.valueOf(getIntProperty("LogNivel"));
    }
    
    public void setLogNivel(NivelLog value) throws Exception {
        this.setIntProperty("LogNivel", value.asInt());
    }
    
    public String getLogPath() throws Exception{
       return getProperty("LogPath");
    }
    
    public void setLogPath(String value) throws Exception {
        this.setProperty("LogPath", value);
    }    
}
